data/mods/Magiclysm/eoc_spell_learn_boost.json

```diff
@@ -0,0 +1,140 @@
+[
+  {
+    "type": "jmath_function",
+    "id": "mc_spell_exp_diff",
+    "num_args": 1,
+    "//": "accept the spell level, return a difference in experience between spell's current level and the next level",
+    "return": "spell_exp(_0 + 1) - spell_exp(_0)"
+  },
+  {
+    "type": "jmath_function",
+    "id": "mc_spell_train_focus_factor",
+    "num_args": 2,
+    "//args": "_0=focus, _1=spell_level",
+    "//": "At level 15 focus factor is enabled, 50 focus = twice more spellcasts, 100 focus = same, 200 focus = twice less spellcasts",
+    "return": "_1 > 15 ? (max(20, _0) - 50) / 100 + 0.5 : 1"
+  },
+  {
+    "type": "jmath_function",
+    "id": "mc_spell_cost_train_factor",
+    "num_args": 3,
+    "//args": "_0=cost,_1=min_spellcasts,_2=max_spellcasts",
+    "//": "accept the spell _cost, return how much spellcasts required to level the spell; formula assumes 10 spellcasts at 100% manapool cost, and 2 spellcasts at 1% manapool cost",
+    "return": "max(_1, min(_2, (_2 - (_1 * max(0, _0) / u_val('mana_max') * _2 / _1 ))))"
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MC_CASTED_A_SPELL_LEARN_BOOST",
+    "eoc_type": "EVENT",
+    "required_event": "character_casts_spell",
+    "//": "Regular classes levels some faster and needs 2-10 spellcasts to level up depends on spell cost",
+    "condition": {
+      "or": [
+        {
+          "compare_string": [ "NONE", { "context_val": "school" } ],
+          "//todo": "Should we check somehow 0 mana cost (for stamina cost spells etc)?"
```

Comment:
_cost return the cost of the spell no matter is it mana or any another type of the energy; we probably should handle animist spells separately since they use rune stones they made previously, and do not use any mana

Reply:
ATM 0 cost spells will get 5% exp (20 spellcasts). Is is okay or should we handle it somehow? Have no idea actually...

Reply:
i think it's okay, we don't have any way to handle it properly except making multiple EoCs, that check each spell id

Reply:
Not every animist's spells are mana free, and not every mana free spells has same flags... Dunno.